ogr/abstract.py

```diff
@@ -168,15 +168,34 @@ def __repr__(self) -> str:
 
 
 class Reaction(OgrAbstractClass):
-    def __init__(self, raw_reaction: Any) -> None:
+    def __init__(
+        self,
+        raw_reaction: Any,
+        reaction_id: Optional[int] = None,
+        parent: Optional[Any] = None,
+    ) -> None:
         self._raw_reaction = raw_reaction
+        self._id = reaction_id
+        self._parent = parent
 
     def __str__(self):
         return f"Reaction(raw_reaction={self._raw_reaction})"
 
     def delete(self) -> None:
         """Delete a reaction."""
-        raise NotImplementedError()
+        if not self._id:
+            raise ValueError("Cannot delete a reaction without an ID.")
```

Comment:
This clashes with `reaction_id` being optional. If the intent is that `id` is always present then there should be an overwritable `@property` with the default being to read `self._id` or raise.